### PR TITLE
SLint-related codestyle changes

### PR DESCRIPTION
**Pull request by akak1977**

### CODE REVIEW

File: src/VirtoCommerce.Platform.Core/PlatformConstants.cs [hunk 1]
```diff
@@ -67,7 +67,7 @@ public static class Settings
         {
             public static class Security
             {
-                public static SettingDescriptor SecurityAccountTypes = new SettingDescriptor
+                public static SettingDescriptor SecurityAccountTypes { get; set; } = new SettingDescriptor
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Core/PlatformConstants.cs]: left only `get;`  accessors for constants.

File: src/VirtoCommerce.Platform.Core/Common/ReflectionUtility.cs [hunk 2]
```diff
@@ -86,7 +85,7 @@ public static bool IsDerivativeOf(this Type type, Type typeToCompare)
         {
             if (type == null)
             {
-                throw new NullReferenceException();
+                throw new ArgumentNullException("type");
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Core/Common/ReflectionUtility.cs]: use `nameof()` instead of string

Comment (akak1977) [reply src/VirtoCommerce.Platform.Core/Common/ReflectionUtility.cs]: OK

File: src/VirtoCommerce.Platform.Core/PlatformConstants.cs (hunk 1 above)
Comment (akak1977) [reply src/VirtoCommerce.Platform.Core/PlatformConstants.cs]: OK

**Pull request merged**